src/main/rx/crsf.c

```diff
@@ -227,15 +234,6 @@ void crsfRxSendTelemetryData(void)
 {
     // if there is telemetry data to write
     if (telemetryBufLen > 0) {
-        // check that we are not in bi dir mode or that we are not currently receiving data (ie in the middle of an RX frame)
-        // and that there is time to send the telemetry frame before the next RX frame arrives
-        if (rxConfig()->halfDuplex) {
-            const uint32_t timeSinceStartOfFrameUs = micros() - crsfFrameStartAtUs;
-            if ((timeSinceStartOfFrameUs < CRSF_TIME_NEEDED_PER_FRAME_US) ||
-                (timeSinceStartOfFrameUs > CRSF_TIME_BETWEEN_FRAMES_US - CRSF_TIME_NEEDED_PER_FRAME_US)) {
-                return;
-            }
-        }
```

Comment:
I've just spotted this change. This block of code has been erroneously removed with no explanation why other than half-duplex scheduling being considered obsolete.

It may be considered obsolete, but there is still a CLI setting to enable it, so the user can enable it and if they do so the code will be broken. This code should not be removed until the ability to set half-duplex is also removed.

Reply:
@martinbudden I believe this ability should be removed from the CLI, as the protocol is full duplex.  Based on the spec, the half-duplex mode is for the communication between the RC and the TX module, rather than the RX to the flight controller.

Reply:
I also removed it after conversation with @mikeller (if I remember correctly) as it was considered to be dead code.  Prior to the CLI command being added, it was not available for configuration, so the code was sitting there and was never used.  I think the timing was just off.

Reply:
Yes. When I came across this, the code for half duplex was dead code, rendered dead in a hard to spot way, so I changed it to be actually used (https://github.com/betaflight/betaflight/commit/860ec23fbe1916d96e458edca498af26cc304fc2), which seemed to be the right thing to do, especially since `rxConfig()->halfDuplex`, causing setting of this parameter to on to render the CRSF RX nonfunctional.
Later on, @codecae pointed out to me that this mode is not supported by current CRSF firmware, and suggested we better remove code that had never been functional.
As it is, the current CRSF code just entirely ignores `rxConfig()->halfDuplex`, which seems to be the right thing to do, since the RX itself does not support half duplex, and is (in my opinion) preferable to putting the flight controller into a mode that will not work with anything, and hence cannot be tested.

Reply:
OK, fair enough.